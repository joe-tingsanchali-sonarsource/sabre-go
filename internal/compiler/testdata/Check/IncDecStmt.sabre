package main

func correct() {
    x := 1
    y := 1.5
    x++
    x--
    y++
    y--
}

func incorrect() {
    x := true
    x++
    true++
    false--
}

func main() {
    correct()
}
