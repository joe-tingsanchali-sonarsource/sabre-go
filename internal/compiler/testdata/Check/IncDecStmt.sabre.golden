>> 	    x++
>> 	    ^^^ 
Error[internal/compiler/testdata/Check/IncDecStmt.sabre:14:5]: expected integral or floating type, but found 'bool'
>> 	    true++
>> 	    ^^^^^^ 
Error[internal/compiler/testdata/Check/IncDecStmt.sabre:15:5]: expression is not assignable
>> 	    false--
>> 	    ^^^^^^^ 
Error[internal/compiler/testdata/Check/IncDecStmt.sabre:16:5]: expression is not assignable
